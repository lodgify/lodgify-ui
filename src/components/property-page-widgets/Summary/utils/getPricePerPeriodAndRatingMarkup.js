import React, { Fragment } from 'react';
import { Segment } from 'semantic-ui-react';

import { getPricePerPeriodMarkup } from 'utils/get-price-per-period-markup';
import { Rating } from 'elements/Rating';
import { TextPlaceholder } from 'elements/TextPlaceholder';

/**
 * @param  {boolean} isShowingPlaceholder
 * @param  {string}  periodText
 * @param  {string}  pricePerPeriod
 * @param  {number}  ratingNumber
 * @return {Object}
 */
export const getPricePerPeriodAndRatingMarkup = (
  isShowingPlaceholder,
  periodText,
  pricePerPeriod,
  ratingNumber
) =>
  isShowingPlaceholder ? (
    <div className="placeholder-wrapper">
      <TextPlaceholder length="medium" />
      <TextPlaceholder length="short" />
    </div>
  ) : (
    <Fragment>
      <Segment>
        {getPricePerPeriodMarkup(pricePerPeriod, periodText, 'small')}
      </Segment>
      {!!ratingNumber && (
        <Segment className="is-rating">
          <Rating iconSize="tiny" ratingNumber={ratingNumber} />
        </Segment>
      )}
    </Fragment>
  );
