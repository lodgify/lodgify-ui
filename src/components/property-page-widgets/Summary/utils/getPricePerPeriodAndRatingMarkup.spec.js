import React from 'react';
import { mount } from 'enzyme';

import { getPricePerPeriodAndRatingMarkup } from './getPricePerPeriodAndRatingMarkup';

const pricePerPeriod = '280$';
const periodText = 'wooble';

const getMarkupAsRenderedComponent = (ratingNumber, isShowingPlaceholder) =>
  mount(
    <div>
      {getPricePerPeriodAndRatingMarkup(
        isShowingPlaceholder,
        periodText,
        pricePerPeriod,
        ratingNumber
      )}
    </div>
  );

describe('getPricePerPeriodAndRatingMarkup', () => {
  describe('if `isShowingPlaceholder` === true', () => {
    it('should return the correct structure', () => {
      const actual = getMarkupAsRenderedComponent(undefined, true);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `ratingNumber` is not `0`', () => {
    it('should return the right markup', () => {
      const ratingNumber = 1;
      const wrapper = getMarkupAsRenderedComponent(ratingNumber);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('if `ratingNumber` is `0`', () => {
    it('should return the right markup', () => {
      const ratingNumber = 0;
      const wrapper = getMarkupAsRenderedComponent(ratingNumber);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
