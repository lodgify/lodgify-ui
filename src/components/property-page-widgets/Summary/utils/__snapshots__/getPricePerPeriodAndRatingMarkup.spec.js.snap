// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPricePerPeriodAndRatingMarkup if \`isShowingPlaceholder\` === true should return the correct structure 1`] = `
<div>
  <div
    className="placeholder-wrapper"
  >
    <TextPlaceholder
      isFluid={true}
      length="medium"
    >
      <div
        className="placeholder text medium fluid"
      />
    </TextPlaceholder>
    <TextPlaceholder
      isFluid={true}
      length="short"
    >
      <div
        className="placeholder text short fluid"
      />
    </TextPlaceholder>
  </div>
</div>
`;

exports[`getPricePerPeriodAndRatingMarkup if \`ratingNumber\` is falsy should return the right markup 1`] = `
<div>
  <Segment>
    <div
      className="ui segment"
    >
      <span>
        from 
        <Heading
          className={null}
          hasMargin={true}
          isColorInverted={false}
          size="small"
          textAlign={null}
        >
          <Header
            as="h4"
            className=""
            inverted={false}
            textAlign={null}
          >
            <h4
              className="ui header"
            >
              280$
            </h4>
          </Header>
        </Heading>
        wooble
      </span>
    </div>
  </Segment>
</div>
`;

exports[`getPricePerPeriodAndRatingMarkup if \`ratingNumber\` is truthy should return the right markup 1`] = `
<div>
  <Segment>
    <div
      className="ui segment"
    >
      <span>
        from 
        <Heading
          className={null}
          hasMargin={true}
          isColorInverted={false}
          size="small"
          textAlign={null}
        >
          <Header
            as="h4"
            className=""
            inverted={false}
            textAlign={null}
          >
            <h4
              className="ui header"
            >
              280$
            </h4>
          </Header>
        </Heading>
        wooble
      </span>
    </div>
  </Segment>
  <Segment
    className="is-rating"
  >
    <div
      className="ui segment is-rating"
    >
      <Rating
        iconSize="tiny"
        isShowingNumeral={true}
        ratingNumber={1}
      >
        1
        <Rating
          clearable="auto"
          disabled={true}
          maxRating={5}
          rating={1}
          size="tiny"
        >
          <div
            className="ui tiny disabled rating"
            onMouseLeave={[Function]}
            role="radiogroup"
          >
            <RatingIcon
              active={true}
              aria-checked={true}
              aria-posinset={1}
              aria-setsize={5}
              as="i"
              index={0}
              key="0"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={true}
                aria-posinset={1}
                aria-setsize={5}
                className="active icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={2}
              aria-setsize={5}
              as="i"
              index={1}
              key="1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={2}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={3}
              aria-setsize={5}
              as="i"
              index={2}
              key="2"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={3}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={4}
              aria-setsize={5}
              as="i"
              index={3}
              key="3"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={4}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={5}
              aria-setsize={5}
              as="i"
              index={4}
              key="4"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={5}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
          </div>
        </Rating>
      </Rating>
    </div>
  </Segment>
</div>
`;
